connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala

```diff
@@ -128,6 +128,7 @@ object KustoWriter {
       val partitionsResults = rdd.sparkContext.collectionAccumulator[PartitionResult]
       val parameters = KustoWriteResource(authentication = authentication, coordinates = tableCoordinates,
         schema = data.schema, writeOptions = rebuiltOptions, tmpTableName = tmpTableName)
+      val sinkStartTime = Instant.now()
```

Comment:
please note that in some clients/pipelines, customers can override the `creationTime` of extents created by ingestion as part of ingestion properties, in which case - the minimum and maximum values for the datetime range should be based on the properties they pass.
if this isn't supported in the Spark connector - then all is good.
if it is, then this will need to be adjusted accordingly (i.e. if `creationTime` is specified - use that instead of `Instants.now()`)

Reply:
Done.

Reply:
i think you misunderstood. let me try to explain differently
- you don't have to change the creationTime property. only in rare cases do customers choose to do so.
- if _the customer_ hasn't chosen to set the 'creationTime' property, then using the current datetime (UTC now) to capture the job's start/end time is just fine.

Reply:
Please correct us here @y0nil . The logic was the following. If the user overrides the extentCreationTime, then we set it this way or else fallback to now. Did we miss something here.

```
  def getCreationTime(ingestionProperties: SparkIngestionProperties, tableCoordinates: KustoCoordinates): Instant = {
    val creationTime = Option(ingestionProperties.toIngestionProperties(tableCoordinates.database, tableCoordinates.table.get).getAdditionalProperties.get("creationTime"))
    creationTime match {
      case Some(creationTimeVal) => Instant.parse(creationTimeVal)
      case None => Instant.now()
```

Reply:
no, you don't need to explicitly set creationTime. only if the end-user specified is should it be set.
otherwise, it shouldn't be explicitly set.

Reply:
Yes. Agreed Yoni. I guess that's what we did here? We are not setting the creationTime anywhere, but just taking the creationTime (only if exists) in first place, else the current time.

Reply:
let me try to explain again - you should never set creationTime to "Instant.now()". you should only set it if it was provided by the end user

Reply:
I think you are confusing the local variable creationTime with the ingestion property `creationTime`. Sorry for the confusing name will change it. 
```
def getCreationTime(ingestionProperties: SparkIngestionProperties, tableCoordinates: KustoCoordinates): Instant = {
    val creationTime = Option(ingestionProperties.toIngestionProperties(tableCoordinates.database, tableCoordinates.table.get).getAdditionalProperties.get("creationTime"))
    creationTime match {
      case Some(creationTimeVal) => Instant.parse(creationTimeVal)
      case None => Instant.now()
```

The local variable, creationTime, will be set if the ingestion prop `creationTime` is provided by user. Else the current Instant will be set. and returned.

Reply:
yes, that's much clearer than the previous version you quoted

```
    creationTime match {
      case Some(creationTimeVal) => Instant.parse(creationTimeVal)
      case None => Instant.now()
```

Reply:
Thanks @y0nil, with the semantic change does it sound okay :) if so , will fold it into master and plan a release this week.